src/control/server/storage_scm.go

```diff
@@ -48,6 +67,166 @@ var (
 	msgScmUpdateNotImpl     = "scm firmware update not supported"
 )
 
+type runCmdFn func(string) ([]byte, error)
+
+// ScmSetup is an interface to configure scm prerequisites via shell tools
+type ScmSetup interface {
+	prep(common.ScmModules) ([]byte, error)
+	reset() error
+}
+
+// scmSetup is an implementation of the ScmSetup interface
+type scmSetup struct {
+	config *configuration
+}
+
+// prep executes commands to configure SCM modules into AppDirect
+// interleaved regions/sets hosting pmem kernel device namespaces.
+//
+// State will be established based on presence of modules, presence of
+// regions and free space on regions.
+//
+// Actions based on state:
+// * no modules -> no-op
+// * modules exist and no regions -> create all regions (needs reboot)
+// * regions exist and free capacity -> create all namespaces
+// * regions exist but no free capacity -> no-op
+//
+// Command output from external tools will be returned.
+func (s *scmSetup) prep(modules common.ScmModules) ([]byte, error) {
+	if len(modules) == 0 {
+		return []byte(msgScmNoModules), nil
+	}
+
+	state, err := getState(run)
+	if err != nil {
+		return []byte{}, errors.WithMessage(err, "establish scm state")
```

Comment:
No need to create an empty byte slice... nil accomplishes the same thing.

Reply:
thanks